### DISCUSSION TITLE
Advice on supporting fragments in a Lit-rendered app?

### DISCUSSION THREAD

**Discussion by jyasskin** [lit/lit#3524] [Q&A]
https://chromestatus.com/feature/6577673212002304 is rendered as a Lit element, which means that links like https://chromestatus.com/feature/6577673212002304#status, which should [scroll](https://html.spec.whatwg.org/multipage/browsing-the-web.html#scrolling-to-a-fragment) directly to the `<section id="status">` element, instead can't find that element because it's embedded in a custom element's shadow DOM.

What's the best practice for making URL fragments work?

**Comment by augustjk** [comment]
Unfortunately I don't know of a super clean solution for this.

You could make it so that the element to scroll to is present in the light DOM and slotted into the appropriate spot within the shadow DOM. It can get a bit messy if you have multiple of those that need to go several layers of custom elements.

For a JS solution, you could write an event listener on the component that renders the element to navigate to such that it listens for the `popstate` event and finds and scroll to the element via `scrollIntoView()`.

Here's a rough example showing both approaches: https://lit.dev/playground/#gist=782fcd691a020c6b2e5347fa81f90ee6
